public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -104,6 +114,36 @@ export function runSplitGroupedQueries(datasource: LokiDatasource, requests: Lok
       done();
     };
 
+    const retry = (errorResponse?: DataQueryResponse) => {
+      if (errorResponse?.errors && errorResponse.errors[0].message?.includes('maximum of series')) {
+        console.log(`Maximum series reached, skipping retry`);
+        return false;
+      } else if (errorResponse?.errors && errorResponse.errors[0].message?.includes('parse error')) {
+        console.warn(`Parse error, skipping retry`);
+        shouldStop = true;
+        return false;
+      }
```

Comment:
I would rather go with a list that explicitly enables retries, that with something that skips retries. Can't we just retry timed out queries, for instance?

Btw, is retrying needed for sharding, or something that can be added separately?

Reply:
Yeah, this is a great call. I'll do that to all the retry implementations.

Reply:
Done.